### DISCUSSION TITLE
VSA 2020 QGIS TWW Project

### DISCUSSION THREAD

**Discussion by urskaufmann** [teksi/wastewater#22] [Show and tell]
Starting with the preconfigured "old" qgep-Project.
Changes to be able to load the project without unknown layers
- delete vl-layers, that exist no more (vl_pump_usage_current, vl_maintenace_event_kind, hydraulic_char_data_pump_usage_current,)
-  rename  vl_infiltration_installation_emergency_spillway  and vl_special_structure_emergency_spillway   to *emergency_overflow
- short vl_overflow_char_kind_overflow_characteristic and vl_overflow_char_overflow_chara:cteristic_digital : char in stead of characteristic
- vw_network_node and vw_network_segment: not able to load the layers in the database into the project!? -> @ponceta @3nids

**Comment by urskaufmann** [comment]
- delete all "vl_" in the vl names. Only the groups have this VL_
- deselect all multiline options with text-fields: only wastewater_structure.records and maintenance_event.result are MText. There will be some MText also in Stammkarten, but there are till now no Stammkarten in the project.
- in all fields with value relation to a vl: add field active as filter, in all fk-fields with value relation to organisation add "status" is not 9048 (is not gone)
- set all fields obj_id and last_modification to not editable
- layer organisation: rename (no od_), add new fields status and organisation_type with there vl. Add other new fields identifer_short, municipality_number, uid, fk_dataowner and fk_provider , configure Drag and Drop-Form

**Comment by urskaufmann** [comment]
- layer pipe_profile: rename (no od_), add fk_dataowner and fk_provider, constraint in field height_width_ratio
- first changes to layer vw_tww_wastewater_structure (not finished yet)

**Comment by 3nids** [comment]
> vw_network_node and vw_network_segment: not able to load the layers in the database into the project

the views are materialized. They probably need to be refreshed: `  REFRESH MATERIALIZED VIEW tww_app.vw_network_node;
  REFRESH MATERIALIZED VIEW tww_app.vw_network_segment;` should do the trick.

**Reply by ponceta** [reply]
Right click on pgAdmin as additionnal information :

![image](https://github.com/teksi/wastewater/assets/8614223/26a9f987-e78c-4465-8194-8a72118aeeed)

**Comment by urskaufmann** [comment]
- catchment_area: 4 relations to wastewater_node to be able to define the connections on the map. stile-definitions for current and planned. symbol colors defined as project-colors. colors taken as possible from Darstellungsmodell canton Bern/Solothurn (https://www.bvd.be.ch/content/dam/bvd/dokumente/de/awa/wasser/abwasserentsorgung/genereller-entw%C3%A4sserungsplan-gep/wegleitung-daten-se-darstellungsmodell-gep-beilage5.pdf and https://github.com/ipw-informationsplattform/dssminipub_public)
new values in vl_drainage_system_current/planned.
- to do: labels for catchment_area
- 4 new layer:  vw_catchment_area_wwc / wwp / rwc / rwp_connections. Grouped in catchment connection lines

**Comment by urskaufmann** [comment]
- new vw_tww_overflow. Relations fk_wastewater_node, fk_wastewater_node_to, fk_overflow_char. Symbol with graphicgenerator (Make_Line($geometry, geometry(get_feature('vw_wastewater_node','obj_id', fk_overflow_to ))) Shows a line between from- and to-node. If vw_wastewater_node has translated name, the symbol has to be translated also...
- relation hqrelation.fk_overflow_char to be able to define the hqrelation

**Comment by urskaufmann** [comment]
- project_colors defined also for channel.usage and for wastewater_structure.renovation_necessity (this colors can also be used for structure_condition and the new fields urgency_figure and condition_score.
- project variables defined for limits of visibilty in the main layers.
- main layers: some symbols and labels for overviewmap and renovation_necessity defined, simplified, without details of the network_plan, that are not necessary for GEP work (e.g. accessability, horizontal_positioning). Similar as in Darstellungsmodell BE/SO.
- Defined map themes for overview and renovation_necessity.

**Comment by urskaufmann** [comment]
version 2024-01-07:
- vw_backflow_prevention: no more in project, use vw_throttle_shut_off_unit instead. -> group unused
- subclass-views of overflow no more in project.  -> group unused
- new overflows should be created with vw_tww_overflow by clicking on node and to_node. new overflow in relation vw_wastewater_node is not active, because it whould then also be created with click on the node which makes not sense, if starting from the node.

- to discuss: vw_throttle_shut_off_unit with geometry of node to visualise and create the points similar as vw_tww_overflow ?!

- duplicate records would be helpful with pumps and throttle_shut_off_unit. But does not work with pumps!!  because when duplicating, the obj_id is not created with the normal sequences but by adding "_1" to the existing obj_id. see also https://github.com/QGEP/QGEP/issues/666   -> how can this be solved ??????

- because double fields in the attribute form have problems with multiedit and with constraints, I propose to delete this doubles and have this fields just on the general-tab. That concerns the two main-views. If QGIS can handle double fields, it's not lot work to add this fields again. I have rearranged the tabs in attribute form for main views

- the order of the fields in the attributetables are rearranged. First always identifier and obj_id, then important fields (in my opinion), other fields grouped by class, dataowner and provider at the end.

- set constraints for numeric field to check to limits in the main-views (other views later). 
- Check max. length of entry for field wastewater_structure.location_name with constraint

**Reply by urskaufmann** [reply]
double fields in attribute form is no more a problem with QGIS 3.28.15 or 3.34.3. 
Question: add the fields from the general tab again in the other tabs to have the fields of one class complete on this tabs?

**Reply by urskaufmann** [reply]
Dont know anymore what I have tested a month ago - now it does not work. Not with gpkg, not with postgresql, not with 3.28.15 and not with 3.34.3
If there is a field twice in the attribut form (1. and 2. tab), it accepts my change only, if I edit on the second tab. On the first tab, there is the red sign, that there is unsaved data, but it seems that QGIS just looks at the second (means last) tab with these field.

**Comment by urskaufmann** [comment]
There is still a path in the project, that should not be there: 2infra/infraprogs/QGIS 3.0/bin
@ponceta : can you correct this please
